Change healing logic

MatthiasMoore opened the pull request:
The player will be healed when they are not full hp during boss battles.

Resolves #177

MatthiasMoore committed:
Change healing logic

The player will be healed when they are not full hp during boss battles.

meepen commented:
Correct me if im wrong - but the healing attack heals everyone, not just you, correct?

MatthiasMoore commented:
Yea, the healing skill does heal others, and it seems the boss will randomly damage one of the ten players with its laser attack.  It would be more efficient to check all players health values, and then apply it in an optimal way, but I was not thinking about that at the time.

MatthiasMoore committed:
Check if allies need heal before using heal.

If more than half of the allies are not full hp, use the heal skill.

MatthiasMoore committed:
Fix a few early morning mistakes

Reference proper variable within game state.
Return false in the event that m_bBossLevel is false
Change check to be greater than or equal to 5 allies not at full health.

MatthiasMoore commented:
The new commits should now prefer healing when half the allies are not full hp.  I will not have time to test it for a few hours, but looking over the logic a few times, it should function properly unless I am overlooking something simple.
